servers/curated-corpus-api/src/admin/resolvers/queries/UrlMetadata/lib.ts

```diff
@@ -0,0 +1,201 @@
+ * @returns formatted date string, or undefined
+ */
+export const deriveDatePublished = (dateRaw: any): string | undefined => {
+  let formattedDate;
+
+  if (dateRaw) {
+    try {
+      const date = new Date(dateRaw);
+
+      // calling new Date() with a non-date string value will not throw, but
+      // it will result in the toString() method returning 'Invalid Date'
+
+      // calling new Date() with a number may or may not work - let's make sure
+      // it's a number that results in the non-default JS date year.
+      if (date.toString() !== 'Invalid Date' && date.getFullYear() !== 1970) {
+        const year = date.getFullYear();
+        const month = (date.getMonth() + 1).toString().padStart(2, '0');
+        const day = date.getDate().toString().padStart(2, '0');
+
+        const hour = date.getHours().toString().padStart(2, '0');
+        const minute = date.getMinutes().toString().padStart(2, '0');
+        const second = date.getSeconds().toString().padStart(2, '0');
+
+        formattedDate = `${year}-${month}-${day} ${hour}:${minute}:${second}`;
+      }
+    } catch (e) {
+      formattedDate = undefined;
+    }
+
+    return formattedDate;
+  }
+};
+
+/**
+ * looks for a publisher name mapping to domain in our internal db. if found,
+ * returns that. if not, returns the normalized domain name.
+ *
+ * @param db PrismaClient
+ * @param url URL for which to look up the publisher
+ * @returns string publisher name, or normalized domain name
+ */
+export const derivePublisher = async (
+  db: PrismaClient,
+  url: string,
+): Promise<string> => {
+  const domainName = getNormalizedDomainFromUrl(url);
+
+  // look up the publisher value from our internal mapping
+  const publisher = await lookupPublisher(db, url);
+
+  return publisher ?? domainName;
+};
+
+/**
+ * use node fetch to hit the external metadata parser.
+ *
+ * broken out into a standalone function for easier mocking in tests.
+ *
+ * @param url URL for which to fetch metadata
+ * @param metadataParserEnpointUrl URL endpoint of the metadata service
+ * @param metadataParserApiKey API key for the metadata service
+ * @returns JSON from the metadata service, or an empty object on failure
+ */
+export const fetchUrlMetadata = async (
+  url: string,
+  metadataParserEnpointUrl: string,
+  metadataParserApiKey: string,
+): Promise<{ [key: string]: any }> => {
+  // Zyte-specific implementation
+  try {
+    const response = await fetch(metadataParserEnpointUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${Buffer.from(
+          metadataParserApiKey + ':',
+        ).toString('base64')}`,
+      },
+      body: JSON.stringify({
+        url: url,
+        article: true,
+        articleOptions: {
+          extractFrom: 'browserHtml',
+        },
+        tags: {
+          environment: config.app.environment,
+          service: config.app.serviceName,
```

Comment:
I wonder whether we could pass the `apollographql-client-name` header value in here to distinguish between prospecting, manual sections, and the scheduler?